git/annex.go

```diff
@@ -146,16 +146,74 @@ func AnnexPull() error {
 		if strings.Contains(sstderr, "Permission denied") {
 			errmsg = "download failed: permission denied"
 		} else if strings.Contains(sstderr, "Host key verification failed") {
+			// Bad host key configured
 			errmsg = "download failed: server key does not match known host key"
 		} else if strings.Contains(sstderr, "would be overwritten by merge") {
-			errmsg = "download failed: local modified or untracked file would be overwritten by download"
-			// TODO: Which file
+			// Untracked local file conflicts with file being pulled
+			errmsg = fmt.Sprintf("download failed: local modified or untracked files would be overwritten by download:\n  %s", strings.Join(parseFilesOverwrite(sstderr), ", "))
+		} else if strings.Contains(sstderr, "unresolved conflict") {
+			// Merge conflict in git files
+			errmsg = fmt.Sprintf("download failed: files changed locally and remotely and cannot be automatically merged (merge conflict):\n %s", strings.Join(parseFilesConflict(string(stdout)), ", "))
+			// abort merge
+			mergeAbort()
 		}
 		err = fmt.Errorf(errmsg)
 	}
+
+	// annex conflicts are resolved automatically and don't produce an error
+	// checking stdout for message
+	sstdout := string(stdout)
+	if strings.Contains(sstdout, "Merge conflict was automatically resolved") {
+		// Merge conflict in annex files (automatically resolved by keeping both copies)
+		err = fmt.Errorf("files changed locally and remotely. Both files have been kept:\n %s", strings.Join(parseFilesAnnexConflict(sstdout), ", "))
```

Comment:
Maybe I am missing something, but is this line not overwriting any error message constructed in the error handling block above?

Reply:
Yes. Good catch. The error block should return at the end. I'm making a few other changes which should make the whole thing easier to read and maintain. Stand by.